apollo-federation/src/merger/merger.rs

```diff
@@ -199,13 +250,617 @@ impl Merger {
     fn post_merge_validations(&mut self) {
         todo!("Implement post-merge validations")
     }
+
+    /// Merge enum type from multiple subgraphs
+    pub(crate) fn merge_enum(
+        &mut self,
+        sources: Sources<&EnumType>,
+        dest: &EnumType,
+    ) -> Result<(), FederationError> {
+        let usage = self.enum_usages.get(dest.name.as_str()).cloned().unwrap_or_else(|| {
+            // If the enum is unused, we have a choice to make. We could skip the enum entirely (after all, exposing an unreferenced type mostly "pollutes" the supergraph API), but
+            // some evidence shows that many a user have such unused enums in federation 1 and having those removed from their API might be surprising. We could merge it as
+            // an "input-only" or as a "input/output" type, but the hints/errors generated in both those cases would be confusing in that case, and while we could amend them
+            // for this case, it would complicate things and doesn't feel like it would feel very justified. So we merge it as an "output" type, which is the least contraining
+            // option. We do raise an hint though so users can notice this.
+            let usage = EnumTypeUsage::Unused;
+            self.error_reporter.add_hint(CompositionHint {
+                code: HintCode::UnusedEnumType.code().to_string(),
+                message: format!(
+                    "Enum type \"{}\" is defined but unused. It will be included in the supergraph with all the values appearing in any subgraph (\"as if\" it was only used as an output type).",
+                    dest.name
+                ),
+            });
+            usage
+        });
+
+        let mut enum_values: IndexSet<Name> = Default::default();
+
+        enum_values.extend(
+            sources
+                .iter()
+                .filter_map(|(_, source)| source.as_ref())
+                .flat_map(|source| source.values.values())
+                .map(|value| value.node.value.clone()),
+        );
+
+        // Merge each enum value
+        for value_name in enum_values {
+            let value_pos = EnumValueDefinitionPosition {
```

Comment:
You can use `EnumTypeDefinitionPosition::value()` here, with `dest`'s `EnumTypeDefinitionPosition`